GMG/Core/Services/ScoreAudioRenderer.swift

```diff
@@ -0,0 +1,120 @@
+//  Copyright © 2025 ADA 4th GMG. All rights reserved.
+
+import AVFAudio
+import Accelerate
+import Foundation
+
+enum ScoreAudioRenderingServiceError: Error {
+    case bufferAllocationFailed
+    case offlineRenderFailed
+}
+
+protocol ScoreAudioRenderer {
+    func renderToAudioFile(score: Score, fileName: String?) throws -> URL
+}
+
+final class DefaultScoreAudioRenderer: ScoreAudioEngineBase, ScoreAudioRenderer {
+
+    // MARK: - Rendering
+
+    func renderToAudioFile(score: Score, fileName: String? = nil) throws -> URL {
+
+        try prepareToExport()
+
+        // Offline rendering용 출력 포맷
+        let outputFormat = engine.mainMixerNode.outputFormat(forBus: 0)
+
+        // Offline rendering 시작
+        let maxFrames: AVAudioFrameCount = 4096
+        try engine.enableManualRenderingMode(
+            .offline,
+            format: outputFormat,
+            maximumFrameCount: maxFrames
+        )
+
+        try engine.start()
+
+        // 출력 파일 URL 설정
+        let outputURL = FileManager.default.temporaryDirectory
+            .appendingPathComponent(fileName ?? score.title)
```

Comment:
score.title만 넘기면 확장자가 빠져서 `fileName`은 확장자 없이 받는 것이 좋을 것 같아요~

```suggestion
        let outputURL = URL.temporaryDirectory.appending(component: "\(fileName ?? score.title).m4a")
```